lib/p5.play.js

```diff
@@ -4338,4 +3909,1314 @@ p5.prototype._warn = function(message) {
   }
 };
 
+  /**
+   * Collision Shape Base Class
+   *
+   * We have a set of collision shapes available that all conform to
+   * a simple interface so that they can be checked against one another
+   * using the Separating Axis Theorem.
+   *
+   * This base class implements all the required methods for a collision
+   * shape and can be used as a collision point with no changes.
+   * Other shapes should inherit from this and override most methods.
+   *
+   * @class p5.CollisionShape
+   * @constructor
+   * @param {p5.Vector} [center] (zero if omitted)
+   * @param {number} [rotation] (zero if omitted)
+   */
+  p5.CollisionShape = function(center, rotation) {
```

Comment:
Okay, here we go - all new code :D

`CollisionShape` is effectively an abstract class, a convenient place to implement the displacement check at the heart of the new collision system and the own/parent transform stuff.